docs/DeveloperGuide.md

```diff
@@ -1,38 +1,619 @@
-# Developer Guide
+# Apollo Developer Guide
+
+## Table of Contents
+
+1. [Acknowledgments](#acknowledgements)
+2. [Design](#design)
+    + [Architecture](#architecture)
+    + [UI Component](#ui-component)
+    + [Logic Component](#logic-component)
+    + [Model Component](#model-component)
+    + [Storage Component](#storage-component)
+    + [Common Classes](#common-classes)
+3. [Implementation](#implementation)
+    + [Add Module](#add-module)
+    + [Delete Module](#delete-module)
+    + [List Modules](#list-modules)
+    + [Show Module](#show-module)
+    + [Add Task](#add-task)
+    + [Delete Task](#delete-task)
+    + [Mark Task As Done](#mark-task-as-done)
+    + [Unmark Task](#unmark-task)
+    + [Find Task](#find-task)
+    + [List Task](#list-task)
+    + [Find Task on Date](#find-task-on-date)
+    + [Storage](#storage)
+    + [Logging](#logging)
+4. [Documentation, logging, testing, configuration, dev-ops](#documentations-logging-testing-configuration-dev-ops)
```

Comment:
I like all the detailed documentation! Perhaps you could include a sequence and class diagram to help with visualisation?